src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -303,11 +302,10 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
       },
 
       getWidgetState(uiState, { searchParameters }) {
-        const page = searchParameters.page || 0;
-
-        if (!hasShowPrevious || !page) {
-          return uiState;
-        }
+        const page = Math.max(
```

Comment:
It used to just return `uiState` (possibly without `page` in it) and let `getWidgetSearchParameters` to use `0` when `page` is missing.
```js
const page = uiState.page ? uiState.page - 1 : 0;
```

It can be incorrect, so now it always return `uiState` with `page`. And it gets the max number between the two, because when the current page is 4, user can click "showPrevious" which gives us smaller page, but we shouldn't change the page to that.